h/notification/reply.py

```diff
@@ -89,8 +89,8 @@ def get_notification(request, annotation, action):
         return
 
     # Don't send reply notifications to the author of the parent annotation if
-    # the author doesn't have permission to read the reply.
-    if not auth.has_permission(request, reply, parent.userid, 'read'):
+    # the reply was private.
+    if not reply.shared:
```

Comment:
It's worth noting that this simplification is valid because of [this code](https://github.com/hypothesis/h/blob/3a2f303177d6a4883bb73d3c3fe23ac027336713/h/api/storage.py#L69-L81), which ensures that replies are always created in the same group as their parent.